src/pat-redactor.js

```diff
@@ -1,121 +1,191 @@
+        default:
+          break;
+      }
 
-        init: function($el, opts) {
-            var i, poptions = parser.parse($el, opts), options = {};
-            if ($.browser.msie && parseInt($.browser.version, 10) === 8) {
-                options.rangy = true;
-            }
+      if (poptions.toolbar.external) {
+        options.toolbarExternal = poptions.toolbar.external;
+      }
 
-            switch (poptions.toolbar.type) {
-                case 'air':
-                    options.air = true;
-                    if (poptions.airButtons.length !==0) {
-                        options.airButtons = poptions.airButtons;
-                    }
-                    break;
-                case 'fixed':
-                    options.toolbarFixed = true;
-                    if (poptions.toolbar['fixed-target']) {
-                        options.toolbarFixedTarget = poptions.toolbar['fixed-target'];
-                    }
-                    break;
-                default:
-                    break;
-            }
-            if (poptions.toolbar.external) {
-                options.toolbarExternal = poptions.toolbar.external;
-            }
-            // Remove dashes from button names, per redactor convention
-            for (i=0; i< poptions.buttons.length; i++) {
-                poptions.buttons[i] = poptions.buttons[i].replace('-', '');
-            }
-            options.buttons = poptions.buttons;
-            options.imageResizable = poptions.image.resizable;
-            options.buttonSource = poptions.showSourceButton;
-            _.extend(options,
-                _.pick(poptions, [
-                    'minHeight',
-                    'fileUpload',
-                    'imageGetJson',
-                    'deniedTags',
-                    'allowedTags',
-                    'formatting',
-                ])
-            );
-            options['formattingTags'] = options.formatting; // XXX: Some versions of redactor uses formattingTags instead of formatting.
-            // XXX Deprecated (see above where parser's arguments are added)
-            if (poptions.imageupload) {
-                options.imageUpload = poptions.imageupload;
-            }
-            if (poptions.imagegetjson) {
-                options.imageGetJson = poptions.imagegetjson;
-            }
+      options.plugins = poptions.plugins;
+      options.buttons = poptions.buttons;
+      options.source = poptions.showSourceButton;
 
-            if (typeof RedactorPlugins === 'undefined') window.RedactorPlugins = {};
-            RedactorPlugins = this.plugins;
-            options['plugins'] = _.keys(this.plugins);
+      _.extend(
+        options,
+        _.pick(poptions, [
+          "minHeight",
+          "formatting",
+          //"deniedTags",
+          //"allowedTags",
+        ])
+      );
 
-            // Until here
-            $el.redactor(this.initializePlugins(options)[0]);
+      //options.limiter = poptions.limitCharacters;
```

Comment:
I thought this option was removed but it is a plugin now. Have to add this later...